scipy/sparse/_base.py

```diff
@@ -1505,7 +1505,15 @@ def getrow(self, i):
 
 class sparray:
     """A namespace class to separate sparray from spmatrix"""
-    pass
+    
+    
+    def __init__(self, *args, **kwargs):
+        if np.isscalar(args[0]):
+            raise ValueError(
+                "scipy sparse array classes do not \
+                    support instantiation from a scalar"
+            )
+        super().__init__(*args, **kwargs)
```

Comment:
The `__init__` methods are pretty spread out across classes already. If it  spreads more is may become hard to tell which init methods gets called during the init process. So, I think in the long run this check might be better placed inside `_sputils.check_shape`.  But that's just my suggestion. I'm ok with other choices.

@perimosocordiae should probably decide.